### PR TITLE
replica install: fix DS restart failure during replica promotion

### PR DESCRIPTION
**Pull request by jcholast**
Since commit b809c6aa10a396e1a597e1a985c3247de7ed4c54, ipa config file is
created before DS certificate is requested, which makes certmonger request
the certificate from the local system rather than the remote master. This
causes the request to fail, as local httpd is not yet configured at the
time of the request.

Move ipa config file creation to its original place to fix the issue.

https://fedorahosted.org/freeipa/ticket/6392

### CODE REVIEW

File: ipaserver/install/server/replicainstall.py [hunk 1]
```diff
@@ -1510,6 +1508,9 @@ def promote(installer):
         if conn.isconnected():
             conn.disconnect()
 
+    # we now have all the information to properly setup server config
+    create_ipa_conf(fstore, config, ca_enabled)
+
```
Comment (stlaz) [inline ipaserver/install/server/replicainstall.py]: Move this to finally block, please.

**Comment by stlaz** [comment]
ACK.

**Comment by stlaz** [comment]
ACK

**Comment by jcholast** [comment]
Pushed to installer-refactoring.

**Pull request closed without merging**